### PR TITLE
[GH-65]

### PR DESCRIPTION
**Pull request by maisnamrajusingh**
#### Summary
Resolves ticket no #65 Allows multiple teams to have the same message without repeating them 
#### Ticket Link
Fixes https://github.com/mattermost/mattermost-plugin-welcomebot/issues/65

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -111,7 +111,7 @@ To accomplish the above, you can specify the following configuration in your `co
                         ]
                     },
                     {
-                        "TeamName": "staff",
+                        "TeamName": "staff"
```
Comment (hanzei) [inline README.md:114]: Unrelated change?

File: README.md [hunk 2]
```diff
@@ -63,7 +63,7 @@ To configure the Welcome Bot, edit your `config.json` file with a message you wa
 
 where
 
-- **TeamName**: The team for which the Welcome Bot sends a message for. Must be the team handle used in the URL, in lowercase. For example, in the following URL the **TeamName** value is `my-team`: https://example.com/my-team/channels/my-channel
+- **TeamName**: The teams for which the Welcome Bot sends a message for. Must be the team handle used in the URL, in lowercase. For example, in the following URL the **TeamName** value is `my-team`: https://example.com/my-team/channels/my-channel . In case of multiple teams, ensure that the comma seperated fields dont have any white space between them. For example `"my-team,my-team-2"` is valid while `"my-team, my-team-2"` is not valid.
```
Comment (hanzei) [inline README.md:66]: Can we be more forgiving and allow a (or multiple) whit spaces before and after the comma.

**Review by hanzei — CHANGES_REQUESTED** [review]
LGTM one question of the parsing

File: README.md (hunk 2 above)
Comment (maisnamrajusingh) [reply README.md:66]: @hanzei  I have pushed a fixe to allow whitespaces.

File: server/command.go [hunk 3]
```diff
@@ -113,7 +129,10 @@ func (p *Plugin) executeCommandList(args *model.CommandArgs) {
 	// Deduplicate entries
 	teams := make(map[string]struct{})
 	for _, message := range wecomeMessages {
-		teams[message.TeamName] = struct{}{}
+		teamNameSlice := strings.Split(message.TeamName, ",")
+		for _, tn := range teamNameSlice {
+			teams[tn] = struct{}{}
+		}
```
Comment (hanzei) [inline server/command.go:135]: Does spaces need to also be trimmed here?

File: server/http_hooks.go [hunk 4]
```diff
@@ -57,12 +58,15 @@ func (p *Plugin) ServeHTTP(c *plugin.Context, w http.ResponseWriter, r *http.Req
...
+			teamNameSlice := strings.Split(wm.TeamName, ",")
+			for _, tn := range teamNameSlice {
+				if data.Team.Name == tn {
+					for _, ac := range wm.Actions {
+						if ac.ActionName == action.Context.Action {
+							p.processActionMessage(*data, action, *ac)
+							p.encodeEphemeralMessage(w, "")
+							return
+						}
 					}
```
Comment (hanzei) [inline server/http_hooks.go:70]: Does spaces need to also be trimmed here?

File: server/command.go [hunk 5]
```diff
@@ -84,11 +84,27 @@ func (p *Plugin) validateCommand(action string, parameters []string) string {
...
 func (p *Plugin) executeCommandPreview(teamName string, args *model.CommandArgs) {
 	found := false
 	for _, message := range p.getWelcomeMessages() {
-		if message.TeamName == teamName {
-			if err := p.previewWelcomeMessage(teamName, args, *message); err != nil {
+		// splits the string and removes white spaces from underlying strings
+		teamNameSlice := strings.Split(message.TeamName, ",")
+		for i := range teamNameSlice {
+			teamNameSlice[i] = strings.TrimSpace(teamNameSlice[i])
+		}
```
Comment (hanzei) [inline server/command.go:105]: Does a helper  liker for  `message` makes sense?

File: server/welcomebot.go [hunk 6]
```diff
@@ -57,7 +57,7 @@ func (p *Plugin) getSiteURL() string {
 	return *config.ServiceSettings.SiteURL
 }
 
-func (p *Plugin) newSampleMessageTemplate(teamName, userID string) (*MessageTemplate, error) {
+func (p *Plugin) newSampleMessageTemplate(teamID string, userID string) (*MessageTemplate, error) {
```
Comment (hanzei) [inline server/welcomebot.go:60]: The changes in this file come from https://github.com/mattermost/mattermost-plugin-welcomebot/pull/74, right? Would you mind dropping them and we try on https://github.com/mattermost/mattermost-plugin-welcomebot/pull/74 go get merged?

**Review by hanzei — CHANGES_REQUESTED** [review]

File: server/welcomebot.go (hunk 6 above)
Comment (maisnamrajusingh) [reply server/welcomebot.go:60]: @hanzei Yes, doing just that. Will update you once done.

File: README.md (hunk 1 above)
Comment (maisnamrajusingh) [reply README.md]: This is related to this change.

File: server/command.go (hunk 3 above)
Comment (maisnamrajusingh) [reply server/command.go:135]: @hanzei  this won't be needed here.

File: server/http_hooks.go (hunk 4 above)
Comment (maisnamrajusingh) [reply server/http_hooks.go:70]: @hanzei  would be a good idea to trim it here. I'm adding a trimmer here.

**Comment by hanzei** [comment]
@maisnamrajusingh Please let me know when this is ready for a re-review

**Comment by maisnamrajusingh** [comment]
> @maisnamrajusingh Please let me know when this is ready for a re-review

yup it is.

**Comment by jasonblais** [comment]
@hanzei @jfrerich quick reminder to help with reviews, thanks!

File: server/command.go [hunk 7]
```diff
@@ -9,7 +9,7 @@ import (
 	"github.com/mattermost/mattermost-server/v5/model"
 )
 
-const commandHelp = `* |/welcomebot preview [team-name] | - preview the welcome message for the given team name. The current user's username will be used to render the template.
+const commandHelp = `* |/welcomebot preview [team-name-1,team-name-2] | - preview the welcome message for the given team name. The current user's username will be used to render the template.
```
Comment (jfrerich) [inline server/command.go:12]:
```suggestion
const commandHelp = `* |/welcomebot preview [team-name-1,team-name-2] | - preview the welcome message for the given team names. The current user's username will be used to render the template.
```

**Review by jfrerich — APPROVED** [review]
One non-blocking nitpick. Approving, but Ben's requests will need to be resolved before merging

**Comment by maisnamrajusingh** [comment]
@hanzei @jfrerich  are there any more changes needed here ? I think I have covered all the requests.

**Comment by jfrerich** [comment]
@maisnamrajusingh, you're good from my end. Thanks!

File: server/welcomebot.go (hunk 6 above)
Comment (hanzei) [reply server/welcomebot.go:60]: These change still seem to come from https://github.com/mattermost/mattermost-plugin-welcomebot/pull/74#discussion_r657143072. Once you dropped them from the file, I'n happy to approve it.

**Review by hanzei — CHANGES_REQUESTED** [review]

**Comment by maisnamraju** [comment]
@jasonblais @jfrerich  @hanzei  I am closing this PR and creating another one that has better logic and no conflicts with the base branch.

**Comment by maisnamrajusingh** [comment]
@jfrerich @hanzei I am closing this pr and I created a new one to fix the logic and overlapping code. Sorry for the inconvenience and delay.
